DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Utility.cs

```diff
@@ -0,0 +1,175 @@
+                return new DateTimeObject()
+                {
+                    DateValue = new DateTimeOffset(year, month, day, hour, minute, second, millisecond, new TimeSpan(timeZoneHour, timeZoneMinute, 0)),
+                    HasTimeZone = true,
+                };
+            }
+            else
+            {
+                return new DateTimeObject()
+                {
+                    DateValue = new DateTimeOffset(year, month, day, hour, minute, second, millisecond, default),
+                    HasTimeZone = false,
+                };
+            }
+        }
+
+        public static string GenerateDicomDateString(DateTimeOffset date)
+        {
+            return date.ToString("yyyyMMdd", CultureInfo.InvariantCulture);
+        }
+
+        public static string GenerateDicomDateTimeString(DateTimeObject date)
+        {
+            if (date == null)
+            {
+                return null;
+            }
+
+            return date.HasTimeZone == true
+                ? date.DateValue.ToString("yyyyMMddHHmmss.ffffffzzz", CultureInfo.InvariantCulture).Replace(":", string.Empty)
+                : date.DateValue.ToString("yyyyMMddhhmmss.ffffff", CultureInfo.InvariantCulture);
+        }
+
+        public static AgeValue ParseAge(string age)
+        {
+            if (string.IsNullOrEmpty(age))
+            {
+                return null;
+            }
+
+            foreach (var item in AgeTypeMapping)
+            {
+                if (new Regex(@"\d{3}" + item.Key).IsMatch(age))
+                {
+                    return new AgeValue(uint.Parse(age.Substring(0, AgeStringLength)), item.Value);
+                }
+            }
+
+            if (uint.TryParse(age, out uint result))
+            {
+                return new AgeValue(result, AgeType.Year);
+            }
+            else
+            {
+                throw new DicomDataException("Invalid age string. The valid strings are nnnD, nnnW, nnnM, nnnY.");
+            }
+        }
+
+        public static string AgeToString(AgeValue age)
+        {
+            if (age == null)
+            {
+                return null;
+            }
+
+            foreach (var item in AgeTypeMapping)
+            {
+                if (age.AgeType == item.Value)
+                {
+                    // Age string only support 3 charaters
+                    if (age.Value <= MaxAgeValue)
+                    {
+                        return age.Value.ToString().PadLeft(AgeStringLength, '0') + item.Key;
+                    }
+                    else if (age.AgeInYears() <= MaxAgeValue)
+                    {
+                        return age.AgeInYears().ToString().PadLeft(AgeStringLength, '0') + item.Key;
+                    }
+
+                    throw new DicomDataException("Invalid age value for DICOM. The valid strings are nnnD, nnnW, nnnM, nnnY.");
+                }
+            }
+
+            return null;
```

Comment:
>null [](http://example.com/codeflow?start=19&length=4)

just curious, can the caller of this function receive a null value or a better return value is EmptyString?

Reply:
caller has considered the situation of receiving null result.

Reply:
> caller has considered the situation of receiving null result.

Who's the caller of a public class method? Is this statement a valid assumption?
Anyway I don't think it makes significant difference to return a null or an empty string.

Reply:
well, it has differences. See the code below:

            var strList = new List<string>();

            string emptyStr = String.Empty;
            strList.Add(emptyStr);

            string nullStr = null;
            strList.Add(nullStr);

            foreach (var str in strList)
            {
                Console.WriteLine(str.Length);
            }

For the empty string, it works well, but for the null string, exception throws. Choosing null or empty depends on their design request.